src/js/gutenberg-plugin.js

```diff
@@ -54,6 +55,130 @@ const ClassifAIToggle = () => {
 	);
 };
 
+/**
+ * Callback function to handle API response.
+ *
+ * @param {Object} resp
+ */
+const buttonClickCallBack = async ( resp ) => {
+	const { select, dispatch } = wp.data;
+	const postId = select( 'core/editor' ).getCurrentPostId();
+	const postType = select( 'core/editor' ).getCurrentPostType();
+	const postTypeLabel =
+		select( 'core/editor' ).getPostTypeLabel() || __( 'Post', 'classifai' );
+
+	if ( resp && resp.terms ) {
+		let updateNeeded = false;
+		const taxonomies = Object.keys( resp.terms );
+		const taxTerms = {};
+		taxonomies.forEach( ( taxonomy ) => {
+			let tax = taxonomy;
+			if ( 'post_tag' === taxonomy ) {
+				tax = 'tags';
+			}
+			if ( 'category' === taxonomy ) {
+				tax = 'categories';
+			}
+
+			const currentTerms =
+				select( 'core/editor' ).getEditedPostAttribute( taxonomy ) ||
+				[];
+			const newTerms = [
+				...currentTerms,
+				...resp.terms[ taxonomy ].map( ( term ) =>
+					Number.parseInt( term )
+				),
+			].filter( ( ele, i, a ) => a.indexOf( ele ) === i );
+
+			if ( newTerms && newTerms.length ) {
+				updateNeeded = true;
+				taxTerms[ tax ] = newTerms;
+			}
+		} );
+
+		if ( updateNeeded ) {
+			// Check for edited values in post.
+			const isDirty = await select( 'core/editor' ).isEditedPostDirty();
+			await dispatch( 'core' ).editEntityRecord(
+				'postType',
+				postType,
+				postId,
+				taxTerms
+			);
+			// If no edited values in post trigger save.
+			if ( ! isDirty ) {
+				await dispatch( 'core' ).saveEditedEntityRecord(
+					'postType',
+					postType,
+					postId
+				);
+			}
+
+			// Display success notice.
+			dispatch( 'core/notices' ).createSuccessNotice(
+				`${ postTypeLabel } ${ __(
+					'classified successfully.',
+					'classifai'
+				) }`,
+				{ type: 'snackbar' }
+			);
+		}
+	}
+};
+
+/**
+ *  Classify Post Button
+ */
+const ClassifAIGenerateTagsButton = () => {
+	const processContent = useSelect( ( select ) =>
+		select( 'core/editor' ).getEditedPostAttribute(
+			'classifai_process_content'
+		)
+	);
+	// Display classify post button only when process content on update is disabled.
+	const enabled = 'no' === processContent ? 'no' : 'yes';
+	if ( 'yes' === enabled ) {
+		return null;
+	}
+
+	const postId = wp.data.select( 'core/editor' ).getCurrentPostId();
+	const postTypeLabel =
+		wp.data.select( 'core/editor' ).getPostTypeLabel() ||
+		__( 'Post', 'classifai' );
+	const buttonText = `${ __( 'Classify', 'classifai' ) } ${ postTypeLabel }`;
```

Comment:
Similar to my earlier comment, this should use `sprintf()`

```suggestion
	/** translators: %s Post type label */
	const buttonText = sprintf( __( 'Classify %s', 'classifai' ), postTypeLabel );
```

Again, it shares the string with the existing PHP version.

https://github.com/10up/classifai/blob/919f3b915d39e12009c7d85888371055e6fb0a22/includes/Classifai/Providers/Watson/NLU.php#L835